blogposts/lsif-indexer.md

```diff
@@ -0,0 +1,221 @@
+---
+title: "Writing an LSIF Indexer"
+author: Uwe Hoffmann
+publishDate: 2019-09-28T10:00-07:00
+tags: [
+  blog
+]
+slug: writing-an-lsif-indexer
+heroImage: https://about.sourcegraph.com/sourcegraph-mark.png
+published: true
+---
+
+At a high level, an LSIF indexer analyzes a collection of input source code files and produces an LSIF data file. This LSIF data file, in turn, is used by an editor or a Sourcegraph instance or some other developer tool to offer functionality like "Jump to Definition", "Show References", hover results with documentation and type information, semantic search, etc.
+
+This article explores what it would take to write an LSIF indexer. Along the way, we will learn a little bit about parsing, walking abstract syntax trees, scopes and the LSIF file format.
```

Comment:
I think this should actually be the first paragraph because it motivates why someone might want to read this post. The second paragraph should explain what LSIF is and possibly link to https://about.sourcegraph.com/blog/code-intelligence-with-lsif before diving into LSIF indexers in particular.

Reply:
good point, done